zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/storage/impl/FileSystemBackupStorage.java

```diff
@@ -166,11 +168,15 @@ public void copyToBackupStorage(File srcFile, File destName) throws IOException
           Paths.get(BackupStorageUtil.constructBackupFilePath(destName.getName(), fileRootPath)),
           StandardCopyOption.REPLACE_EXISTING);
     } finally {
-      if (inputStream != null) {
-        inputStream.close();
-      }
-      if (outputStream != null) {
-        outputStream.close();
+      try {
+        if (inputStream != null) {
+          inputStream.close();
+        }
+        if (outputStream != null) {
+          outputStream.close();
+        }
+      } catch (Exception e) {
+        // empty catch block. This is added incase exception occurs before releasing sharedLock.
```

Comment:
@mgao0 Basically the bug was if any exception is raised in this finally block before releasing lock for ex. inputStream.close() raises an exception then lock won't released. And this was the main bug reported by spotbugs.

@desaikomal thanks for the suggestion it worked! I don't need to suppress it!

Reply:
Makes sense. Thanks for explaining.